src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -875,6 +885,36 @@ protected void mergeField(Tokenizer tokenizer,
             mergeField(tokenizer, extensionRegistry, builder);
         }
     }
+    
+    private void handleUnknownField(Tokenizer tokenizer,    										
+    		Message.Builder builder,
+    		int fieldNumber) throws ParseException {    	
+    	tokenizer.tryConsume(":");
+    	if ("{".equals(tokenizer.currentToken())) {
+    		// Message structure
+    		tokenizer.consume("{");
+    		do {
+    			tokenizer.consumeIdentifier();
+    			handleUnknownField(tokenizer, builder, fieldNumber);

```

Comment:
JsonFormat prints unknown message fields as ByteStrings, so there is no need to recurse. However, for groups, JsonFormat does print the entire nested structure, so in this case we'll need to recurse. This will be part of my new commit.